priam/src/main/java/com/netflix/priam/aws/SDBInstanceData.java

```diff
@@ -155,13 +180,17 @@ public void createInstance(PriamInstance instance) throws AmazonServiceException
      *
      * @throws AmazonServiceException
      */
-    public void registerInstance(PriamInstance instance) throws AmazonServiceException {
+    public void registerInstance(PriamInstance instance, String expectedPreviousInstanceId) throws AmazonServiceException {
         logger.info("Registering PriamInstance in SimpleDB: {}", instance);
         AmazonSimpleDB simpleDBClient = getSimpleDBClient();
         PutAttributesRequest putReq = new PutAttributesRequest(sdbDomain, getKey(instance), createAttributesToRegister(instance));
         UpdateCondition expected = new UpdateCondition();
         expected.setName(Attributes.INSTANCE_ID);
-        expected.setExists(false);
+        if (expectedPreviousInstanceId == null) {
+            expected.setExists(false);
+        } else {
+            expected.setValue(expectedPreviousInstanceId);
```

Comment:
According to this documentation: http://docs.aws.amazon.com/AmazonSimpleDB/latest/DeveloperGuide/ConditionalPut.html you only need to have one or the other; "exists=true" is implied by the fact that the field has a value.